functions/collectionManager.ts

```diff
@@ -0,0 +1,121 @@
+const AWS = require('aws-sdk');
+const isTest = process.env.JEST_WORKER_ID;
+const config = {
+  convertEmptyValues: true,
+  ...(isTest && {
+    endpoint: 'localhost:8000',
+    sslEnabled: false,
+    region: 'local-env',
+  }),
+};
+const docClient = new AWS.DynamoDB.DocumentClient(config);
+const table = (isTest) ? process.env.DYNAMO_DB_TEST_TABLE : process.env.DYNAMO_DB_GAME_TABLE;
+import { Game } from "./game";
+import { Collection } from "./collection";
+import { CollectionError } from "./collectionErrorHandler";
+
+export async function getCollection(userID: string, collectionType: string) : Promise<Collection> {
+    let sortKey = `[${userID}]#[collection]#[${collectionType}]`;
+    let params = {
+      TableName: table,
+      Key: {
+        userID: userID,
+        sortKey: sortKey
+      },
+      KeyConditionExpression: `sortKey = ${sortKey} AND userID = ${userID}`,
+      ConditionExpression: 'attribute_exists(sortKey) and attribute_exists(userID)'
+    }
+    
+    try {
+      let response = await docClient.get(params).promise();
+      let collection = serializeDynamoCollectionResponse(response.Item);
+      return collection;
+    } catch (err: any) {
+      //Dynamo returns an empty object if the get can't find a record.
+      //Not sure how to handle this since the documentClient doesn't throw an error
+      if (err.message == "Cannot read property 'userID' of undefined") {
+        throw new CollectionError("Unable to find collection", 404);
+      }
+      throw err
+    }
+}
+
+export async function addGameToCollection(game: Game, collection: Collection) : Promise<Collection> {
+    collection.addGame(game);
```

Comment:
what happens if a user tries to add two games in quick succession? Quick enough that two instances of 'addGameToCollection' were executing at the same time attempting to add two different games.